packages/frontend/src/components/VariablesList/index.tsx

```diff
@@ -0,0 +1,68 @@
+import type { ComponentType } from 'react'
+import List from '@mui/material/List'
+import ListItem, { ListItemProps } from '@mui/material/ListItem'
+import ListItemButton, {
+  ListItemButtonProps,
+} from '@mui/material/ListItemButton'
+import ListItemText from '@mui/material/ListItemText'
+import { Variable } from 'helpers/variables'
+
+function makeListItemComponent(
+  onClick?: (variable: any) => void,
+): ComponentType<ListItemButtonProps & ListItemProps> {
```

Comment:
Actually i made a mistake here, the right return type is `ComponentType<ListItemButtonProps> | ComponentType<ListItemProps>`